### ISSUE TITLE
What is the expected output?

### ISSUE THREAD

**Issue by YanjieGuo** [neu-vi/ezflow#290]
I am testing the result of the pretrained RAFT model with "raft_kubric_step100k.pth". It doesn't seem to be accurate.
![RAFT_output](https://github.com/neu-vi/ezflow/assets/146278327/88e5d7a0-8970-4a27-8f48-2f09a83658a6)

My code and the images are attached here:
[RAFT test.zip](https://github.com/neu-vi/ezflow/files/12743361/RAFT.test.zip)
```
from ezflow.models import Predictor
import torch
from torchvision import io
import matplotlib.pyplot as plt
import numpy as np

def preprocessImg(imgStr):
    if type(imgStr) == str:
        img = io.read_image(imgStr)
    img = img.float()
    img = torch.unsqueeze(img,dim=0)
    return img

def showImgFlow(img,flow,step = 5, scale = 255):
    v=flow.squeeze().permute(1,2,0).detach().numpy()
    x = np.arange(0, v.shape[0], step)
    y = np.arange(0, v.shape[1], step)
    vx = v[::step, ::step, 0]
    vy = v[::step, ::step, 1]

    img = img.squeeze().permute(1,2,0)/scale

    plt.imshow(img)
    plt.quiver(y,x,vy,-vx)
    plt.show()

predictor = Predictor("RAFT", default=True,
    model_weights_path="raft_kubric_step100k.pth"
)

img1 = preprocessImg("rgb1.png")
img2 = preprocessImg("rgb2.png")
flow = predictor(img1, img2)

showImgFlow(img1,flow,step = 5,scale = 255)

print("Done!")
```
Frame 1:
![image](https://github.com/neu-vi/ezflow/assets/146278327/0a30fd1a-ed80-4524-b084-8e240429004a)
Frame 2:
![image](https://github.com/neu-vi/ezflow/assets/146278327/fc0a1e1e-cc64-4eb2-85bd-eca132c28e03)

I have some questions:
1. Is this result expected? Is there a simple example that I can use to verify the result? 
2. What image format should I use? I cannot use the image name directly as the input to the predictor, or it causes a data type issue for the padder. I have to convert the images from uint8 to float so that they are accepted.
3. What should be the image dimension? I saw that it takes `img = torch.randn(2, 3, 256, 256)`. Are the dimensions for (batch, channel, height, width)? I need to add a dimension for a single image such that it is of (1,3,H,W) for it to be accepted. Also, gray scale image with one channel seems to cause a dimension issue for one conv layer.

**Comment by AruniRC** [comment]
I would add some more info here - we expect large displacement vectors to be around the moving object, in this case, the car in the frame. This is why the visualized vectors seem unusual. Thanks!

**Comment by prajnan93** [comment]
1. 
The RAFT model requires images to be normalized in the range `[-1,1]`.  This can be done manually using the equation `2 * (img / 255.0) - 1.0` or by passing a `torchvision.transforms.Normalize` instance to the `Predictor` constructor with mean = `(127.5, 127.5, 127.5)` and std = `(127.5, 127.5, 127.5)`
` 
```
import torchvision.transforms as transforms

predictor = Predictor(
    "RAFT", 
    default=True,
    model_weights_path="raft_kubric_step100k.pth",
    data_transform=transforms.Normalize(mean=(127.5, 127.5, 127.5), std=(127.5, 127.5, 127.5))
)
```
The output flow for the attached images:

### Dense Flow
`pip install flow_vis`  or install directly from [OpticalFlow_Visualization](https://github.com/tomrunia/OpticalFlow_Visualization)
```
import flow_vis
flow_color = flow_vis.flow_to_color(flow_uv, convert_to_bgr=False)
```
![dense_flow](https://github.com/neu-vi/ezflow/assets/89991031/4e0cd9ec-b740-490c-9a8d-a9c414a70630)

### Flow vectors
I had to slightly modify the function `showImgFlow` attached in the zip to view the flow vectors properly. 
`plt.quiver(y,x,vx,-vy,scale=4*step, color="white")`
`showImgFlow(img1,flow,step = 10, scale = 255)`
![flow_vectors](https://github.com/neu-vi/ezflow/assets/89991031/20666a50-b0c5-4c5f-b49d-fe21d2033f46)
___

2.  `torchvision.io.read_image` supports `JPEG ` and `PNG`. I have created a separate PR (#291) to handle image file paths directly in the `Predictor`. Thanks for catching this. 

___

3. Yes the the model accepts images of shape `(batch, channel, height, width)` and requires 3 channels as input. Grayscale images can be converted to RGB by repeating the values along the channel dimension. For example: `img = np.tile(img[..., None], (1, 1, 3))`
___

Let me know if the solutions above work on your end so that I can close this ISSUE.

**Comment by YanjieGuo** [comment]
Thank you so much for your response! This works on my end. It would be nice to have a short example and more explanations in the doc.
